judge/models/problem.py

```diff
@@ -231,13 +242,16 @@ def is_accessible_by(self, user, skip_contest_problem_check=False):
 
         # If the user can edit the problem.
         # We are using self.editor_ids to take advantage of caching.
-        if self.is_editable_by(user) or user.profile.id in self.editor_ids:
+        if self.is_editable_by(user) or user.profile.id in self.editor_ids or self.is_suggester(user):
```

Comment:
` or self.is_suggester(user)` can be removed since `user.profile.id in self.editor_ids` will be True if the user is the problem's suggester